packages/cosmic-swingset/src/launch-chain.js

```diff
@@ -88,6 +88,20 @@ const parseUpgradePlanInfo = (upgradePlan, prefix = '') => {
  *   `chainStorageEntries: [ ['c.o', '"top"'], ['c.o.i'], ['c.o.i.n', '42'], ['c.o.w', '"moo"'] ]`).
  */
 
+/**
+ * @typedef {'leftover' | 'forced' | 'high-priority' | 'intermission' | 'queued'} CrankerPhase
+ *   - leftover: work from a previous block
+ *   - forced: work that claims the entirety of the current block
+ *   - high-priority: queued work the precedes timer advancement
+ *   - intermission: needed to note state exports and update consistency hashes
+ *   - queued: queued work the follows timer advancement
+ */
+
+/**
+ * @typedef {(phase: CrankerPhase) => Promise<boolean>} Cranker runs the kernel
```

Comment:
I'm not sure how I feel about the `Cranker` name. It controls a macro "swingset run", which itself is composed of many cranks. Does that make a vat delivery a "turner"?

Reply:
Hm, yeah, the existing term that this relates to is a "run" or a "swingset run", meaning one invocation of `controller.run()`. Each cosmos block triggers a certain number of runs (one catchup, then maybe some high-priority, then maybe one timer, then maybe some low-priority, then maybe one cleanup).

I think `Runner` is disqualified (too generic), but I'm ok with `Cranker` as a second choice.

Reply:
This really needed a name, and "Runner" is just too generic. I suppose "Turner" would therefore accurately describe a function that performs a single vat delivery, although if we ever need to name that we'll probably find something better.